File: decompiled/4.0.2/be/ehealth/businessconnector/therlink/domain/responses/Acknowledge.java
package be.ehealth.businessconnector.therlink.domain.responses;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;
import org.apache.commons.lang.builder.ToStringBuilder;

public class Acknowledge {
   private boolean isComplete;
   private List<TherapeuticLinkResponseError> listOfErrors;

   public Acknowledge() {
   }

   public boolean isComplete() {
      return this.isComplete;
   }

   public void setComplete(boolean isComplete) {
      this.isComplete = isComplete;
   }

   public List<TherapeuticLinkResponseError> getListOfErrors() {
      if (this.listOfErrors == null) {
         this.listOfErrors = new ArrayList();
      }

      return this.listOfErrors;
   }

   public void setListOfErrors(List<TherapeuticLinkResponseError> listOfErrors) {
      this.listOfErrors = listOfErrors;
   }

   public String toString() {
      ToStringBuilder builder = new ToStringBuilder(this);
      builder.append("Complete? : " + this.isComplete);
      if (this.listOfErrors != null) {
         Iterator<TherapeuticLinkResponseError> it = this.listOfErrors.iterator();

         while(it.hasNext()) {
            TherapeuticLinkResponseError next = (TherapeuticLinkResponseError)it.next();
            builder.append("[");
            builder.append("Error code : " + next.getErrorCode());
            builder.append("ErrorDescription : " + next.getErrorDescription());
            builder.append("]");
            if (it.hasNext()) {
               builder.append(", ");
            }
         }
      }

      return builder.toString();
   }

   public static class Builder {
      private Acknowledge ack = new Acknowledge();

      public Builder() {
      }

      public Builder withComplete(boolean complete) {
         this.ack.setComplete(complete);
         return this;
      }

      public Builder addError(TherapeuticLinkResponseError error) {
         this.ack.getListOfErrors().add(error);
         return this;
      }

      public Acknowledge build() {
         return this.ack;
      }
   }
}

File: src/main/kotlin/org/taktik/connector/business/therlink/domain/responses/Acknowledge.kt
package org.taktik.connector.business.therlink.domain.responses

import java.util.ArrayList
import org.apache.commons.lang.builder.ToStringBuilder

class Acknowledge {
	var isComplete: Boolean = false
	var listOfErrors: MutableList<TherapeuticLinkResponseError> = ArrayList()

	override fun toString(): String {
		val builder = ToStringBuilder(this)
		builder.append("Complete? : " + this.isComplete)
		if (this.listOfErrors != null) {
			val it = this.listOfErrors!!.iterator()

			while (it.hasNext()) {
				val next = it.next()
				builder.append("[")
				builder.append("Error code : " + next.errorCode!!)
				builder.append("ErrorDescription : " + next.errorDescription!!)
				builder.append("]")
				if (it.hasNext()) {
					builder.append(", ")
				}
			}
		}

		return builder.toString()
	}

	class Builder {
		private val ack = Acknowledge()

		fun withComplete(complete: Boolean): Acknowledge.Builder {
			this.ack.isComplete = complete
			return this
		}

		fun addError(error: TherapeuticLinkResponseError): Acknowledge.Builder {
			this.ack.listOfErrors.add(error)
			return this
		}

		fun build(): Acknowledge {
			return this.ack
		}
	}
}
